tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php

```diff
@@ -273,4 +289,90 @@ public function testListTableDateTypeColumns()
         $this->assertSame('datetime', $columns['col_datetime']->getType()->getName());
         $this->assertSame('datetimetz', $columns['col_datetimetz']->getType()->getName());
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createLargeNumberOfTables(): int
+    {
+        // Create a base table for test.
+        $sql = "CREATE TABLE tbl_test_2766_0 (x_id VARCHAR2(255) DEFAULT 'x' NOT NULL, x_data CLOB DEFAULT NULL NULL, x_number NUMBER(10) DEFAULT 0 NOT NULL, PRIMARY KEY(x_id))";
+        $this->_conn->executeUpdate($sql);
+
+        // Create a large number of tables with indexes and foreign keys.
+        for ($i = 1; $i < 649; $i++) {
+            $sql = "CREATE TABLE tbl_test_2766_$i (x_id VARCHAR2(255) DEFAULT 'x' NOT NULL, x_data CLOB DEFAULT NULL NULL, x_number NUMBER(10) DEFAULT 0 NOT NULL, x_parent_id VARCHAR2(255) DEFAULT 'x' NOT NULL, CONSTRAINT tbl_test_2766_fk_$i FOREIGN KEY (x_parent_id) REFERENCES tbl_test_2766_0(x_id), PRIMARY KEY(x_id))";
+            $this->_conn->executeUpdate($sql);
+            $sql = "CREATE UNIQUE INDEX tbl_test_2766_uix_$i ON tbl_test_2766_$i (x_number)";
+            $this->_conn->executeUpdate($sql);
+        }
+
+        // Create a table with quoted identifiers.
+        $sql = "CREATE TABLE \"tbl_testQ_2766_649\" (\"Q_id\" VARCHAR2(255) DEFAULT 'x' NOT NULL, \"Q_data\" CLOB DEFAULT NULL NULL, \"Q_number\" NUMBER(10) DEFAULT 0 NOT NULL, \"Q_parent_id\" VARCHAR2(255) DEFAULT 'x' NOT NULL, CONSTRAINT \"tbl_testQ_2766_fk_649\"  FOREIGN KEY (\"Q_parent_id\") REFERENCES tbl_test_2766_0(x_id), PRIMARY KEY(\"Q_id\"))";
+        $this->_conn->executeUpdate($sql);
+        $sql = "CREATE UNIQUE INDEX \"tbl_testQ_2766_uix_649\" ON \"tbl_testQ_2766_649\" (\"Q_number\")";
+        $this->_conn->executeUpdate($sql);
+
+        return 650;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function checkLargeNumberOfTables(Schema $schema): void
```

Comment:
Not sure which requirements these checks cover. If it's about handling different types, then it should be a set of smaller tests per type. Otherwise, a failing assertion on one column will prevent the ones for the rest of the columns to be performed.

In any event, if it's not directly about performance optimization but you think having them is important, please move them to a separate issue.